### PR TITLE
setting state.on to the right value after sending state changes to the bridge #38

### PR DESCRIPTION
**Pull request by willie68**
THis is the simple solution. At the moment the response of the bridge will not be evaluated. (Maybe in the next step)

### CODE REVIEW

**Comment by amimof** [comment]
@willie68 Thanks for contributing 👏 Please tell me when this is ready for review

**Comment by willie68** [comment]
I think review can happen now. 
I don't know why travis can't build this...

**Comment by amimof** [comment]
How about if `Bri` is set to `0`, i guess the state.on would always be true in any case?

File: group.go [hunk 1]
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
...
+	g.State.On = true
+	return nil
+}
+
+// Ct sets the light color temperature state property
+func (g *Group) Col(new color.Color) error {
+	return g.ColContext(context.Background(), new)
+}
+
+// CtContext sets the light color temperature state property
```
Comment (amimof) [inline group.go:215]: Should be `ColContext` instead of `CtContext`

File: light.go [hunk 2]
```diff
@@ -246,3 +276,31 @@ func (l *Light) AlertContext(ctx context.Context, new string) error {
 	l.State.Effect = new
 	return nil
 }
+
+func ConvertRGBToXy(newcolor color.Color) ([]float32, uint8) {
```
Comment (amimof) [inline light.go:280]: Needs doc comment. See other public functions for reference.

File: light.go [hunk 3]
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
...
+	l.State.On = true
+	return nil
+}
+
+// Ct sets the light color temperature state property
+func (l *Light) Col(new color.Color) error {
+	return l.ColContext(context.Background(), new)
+}
+
+// CtContext sets the light color temperature state property
```
Comment (amimof) [inline light.go:210]: Should be `ColContext` instead of `CtContext`

File: light.go (hunk 2 above)
Comment (amimof) [reply light.go:280]: Please add a unit test for this function

File: light.go (hunk 3 above)
Comment (amimof) [reply light.go:210]: Also needs unit testing

File: group.go (hunk 1 above)
Comment (amimof) [reply group.go:215]: Needs tests

File: group.go [hunk 4]
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
...
+func (g *Group) ColContext(ctx context.Context, new color.Color) error {
```
Comment (amimof) [inline group.go:216]: Needs tests

File: light.go [hunk 5]
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
 		return err
 	}
 	l.State.Ct = new
+	l.State.On = true
+	return nil
+}
+
+// Ct sets the light color temperature state property
+func (l *Light) Col(new color.Color) error {
```
Comment (amimof) [inline light.go:206]: Needs tests

File: light.go [hunk 6]
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
 		return err
 	}
 	l.State.Ct = new
+	l.State.On = true
+	return nil
+}
+
+// Ct sets the light color temperature state property
```
Comment (amimof) [inline light.go:205]: Fix this comment

File: group.go [hunk 7]
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
 		return err
 	}
 	g.State.Ct = new
+	g.State.On = true
+	return nil
+}
+
+// Ct sets the light color temperature state property
```
Comment (amimof) [inline group.go:210]: Update comment

File: group.go [hunk 8]
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
...
+func (g *Group) Col(new color.Color) error {
```
Comment (amimof) [inline group.go:211]: Needs tests

**Comment by willie68** [comment]
Changed the comments, added comment on ConvertRGB, adding tests.
Beside the ConvertRGBToXy Test, all tests are failing on my side. 
(Maybe because the Hue is not responding, because it's off. :-) )

**Comment by amimof** [comment]
@willie68 Did you read my latest comment about `Bri` being `0` and what the expected behavior would be?

**Comment by willie68** [comment]
I just tested with bri = 0 and the light are not off, it's the lowest possible brightness but it's not off. So setting lights to bright 0 will not shut them down. That means state.on = true even if the bri is 0.

File: light_test.go [hunk 9]
```diff
@@ -341,6 +371,7 @@ func TestSetLightAlert(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	assert.True(t, light.IsOn())
```
Comment (amimof) [inline light_test.go:374]: This line is why tests fails. It doesn't receive expected result. Can you have a look?

**Comment by willie68** [comment]
Just missed a single light.State.On = true

**Pull request merged**